src/main/java/gregtech/common/inventory/itemsource/NetworkItemInfo.java

```diff
@@ -26,39 +27,23 @@ public ItemStackKey getItemStackKey() {
         return itemStackKey;
     }
 
-    int extractItem(int amount, boolean simulate) {
-        int amountToExtract = amount;
-        for (ItemSource itemSource : inventories.keySet()) {
-            amountToExtract -= itemSource.extractItem(itemStackKey, amountToExtract, simulate);
-            if (amountToExtract == 0) break;
-        }
-        int extracted = amount - amountToExtract;
-        if (!simulate && extracted > 0) {
-            recomputeItemAmount();
-        }
-        return extracted;
+    void addToSource(ItemSource itemSource, int amount) {
+        inventories.computeIfPresent(itemSource, (key, value) -> value + amount);
+        inventories.putIfAbsent(itemSource, amount);
+        totalItemAmount += amount;
     }
 
-    boolean addInventory(ItemSource inventory, int amount) {
-        if (inventories.getOrDefault(inventory, 0) != amount) {
-            this.inventories.put(inventory, amount);
-            return recomputeItemAmount();
+    void removeFromSource(ItemSource itemSource, int amount) {
+        inventories.computeIfPresent(itemSource, (key, value) -> {
+            if (amount >= value) {
```

Comment:
Wouldn't we only want to do this if `value` is equal to zero? If we can extract a partial stack from an inventory, ie, amount > value && value != 0, wouldn't we want to extract that amount and then gather the remaining from the next source?

Reply:
This updates the map with the amount that was actually removed. If the amount extracted was bigger or equal than the available the key is also removed from the map